samples/matter/common/src/bt_nus_service.cpp

```diff
@@ -0,0 +1,219 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "bt_nus_service.h"
+
+#include <zephyr/logging/log.h>
+#include <zephyr/settings/settings.h>
+
+#include <platform/CHIPDeviceLayer.h>
+
+using namespace ::chip;
+using namespace ::chip::DeviceLayer;
+
+LOG_MODULE_DECLARE(app, CONFIG_MATTER_LOG_LEVEL);
+
+#define NUS_BT_NAME (CONFIG_BT_DEVICE_NAME "_NUS")
+
+namespace
+{
+constexpr uint32_t kAdvertisingOptions = BT_LE_ADV_OPT_CONNECTABLE;
+constexpr uint8_t kAdvertisingFlags = BT_LE_AD_GENERAL | BT_LE_AD_NO_BREDR;
+constexpr uint8_t kBTUuid[16] = { 158, 202, 220, 36, 14, 229, 169, 224, 147, 243, 163, 181, 1, 0, 64, 110 };
```

Comment:
So other examples use `BT_UUID_NUS_VAL`. Isn't that possible?

Reply:
There is something different with the C++ compiler rather than C regarding BT_ macros. When I used `BT_UUID_NUS_VAL` the UUID didn't appear on the mobile phone, and the compiler returns many warnings. So I decided to generate it and provide it in raw format.

Reply:
But wasn't it caused more by `BT_DATA_BYTES` which uses C compound literals that are not supported in C++. Wouldn't `constexpr uint8_t kBTUuid[] = { BT_UUID_NUS_VAL };` work either?

Reply:
hmm, ok, it can work this way, will check.